### ISSUE TITLE
Scribe-Android Outreach 👋🚀

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-Android#62]
<a><img src="https://raw.githubusercontent.com/scribe-org/Organization/main/logo/ScribeLogoRounded.png" width="30%" alt="Scribe Logo" align="right"></a>

## Sharing and promoting Scribe-Android

This pinned issue is for discussing where and how best to promote Scribe-Android. The purpose is to spread the word so that others can find Scribe-Android as potential users, to provide feedback and to grow the contributors base. The issue text will be edited as the discussion below progresses :)

Everyone is welcome to share their input, and efforts to share Scribe-Android are much appreciated! Communities revolving around open-source, open-data, Wikidata, Wikimedia, Kotlin, Android, language learning and the languages Scribe-Android supports should all be considered.

### Where Scribe-Android is currently posted:

- [Wikimedia New Developers page](https://www.mediawiki.org/wiki/New_Developers)

### Organizations that have featured Scribe-Android:

- [Wikimedia](https://www.wikimedia.org/): new developers page and hackathons
- [Wikimedia Deutschland](https://www.wikimedia.de/): Wikidata events, blogposts and social media

### General

- [Mastodon](https://wikis.world/@scribe) - suggested shares and posts welcome!
- [Hacker News](https://news.ycombinator.com/)
- [Product Hunt](https://www.producthunt.com/)

### Conferences

- WIP

### Language and learning institutions

- Forums for places where people learn programming as new developers are welcome
- Major language institutions for a given language
    - For example the [Goethe-Institut](https://www.goethe.de/de/index.html), [Gesellschaft für deutsche Sprache](https://gfds.de/), [Académie française](https://www.academie-francaise.fr/), [Institut de France](https://www.institutdefrance.fr/), etc
    - See [List of language regulators on Wiki](https://en.wikipedia.org/wiki/List_of_language_regulators)
    - See [Higher education institutions that teach Swift](https://developer.apple.com/learn/institutions/)
- University language learning departments for the languages that Scribe-Android supports
- Podcasts and other groups that support/serve language learners
- Institutions that are trying to preserve languages as a Scribe-Android keyboard could be made for them

### Tech subreddits

|                                                        |                                                  |                                            |                                                      |
| :----------------------------------------------------- | :----------------------------------------------- | :----------------------------------------- | :--------------------------------------------------- |
| [r/programming](https://www.reddit.com/r/programming/) | [r/software](https://www.reddit.com/r/software/) | [r/opensource](https://www.reddit.com/r/opensource/) | [r/apple](https://www.reddit.com/r/apple/) | 
| [r/Android](https://www.reddit.com/r/Android/)                 | [r/Kotlin](https://www.reddit.com/r/Kotlin/)       |

### Language subreddits

|                                                                  |                                                        |                                                |                                                          |
| :--------------------------------------------------------------- | :----------------------------------------------------- | :--------------------------------------------- | :------------------------------------------------------- |
| [r/languagelearning](https://www.reddit.com/r/languagelearning/) | [r/learnfrench](https://www.reddit.com/r/learnfrench/) | [r/French](https://www.reddit.com/r/French/)   | [r/German](https://www.reddit.com/r/German/)             |
| [r/italianlearning](https://www.reddit.com/r/italianlearning/)   | [r/Portuguese](https://www.reddit.com/r/Portuguese/)   | [r/russian](https://www.reddit.com/r/russian/) | [r/learnspanish](https://www.reddit.com/r/learnspanish/) |
| [r/Spanish](https://www.reddit.com/r/Spanish/)                   | [r/Svenska](https://www.reddit.com/r/Svenska/)         |

**Comment by andrewtavis** [comment]
@angrezichatterbox, bringing this to your attention in case you have ideas! We do similar issues for Scribe-iOS and Scribe-Data :)

**Comment by angrezichatterbox** [comment]
> @angrezichatterbox, bringing this to your attention in case you have ideas! We do similar issues for Scribe-iOS and Scribe-Data :)

I don't have any specific idea for now but if any come up I'll share it here. 😄

**Comment by shuvashish76** [comment]
The best way to promote it is to show how it can be useful :)
[F-Droid](https://f-droid.org/en/about/) _(a non-profit volunteer project)_ is one of best platform for Android FOSS apps.
If you've any queries feel free to ask in their [forum](https://forum.f-droid.org) or [matrix](https://matrix.to/#/#fdroid:f-droid.org) room :)

**Comment by shuvashish76** [comment]
Join Hacktoberfest : https://hacktoberfest.com/ | https://github.com/topics/hacktoberfest

_(3rd party project which tracks all the android FOSS apps from F-Droid and [IzzyOnDroid](https://apt.izzysoft.de/fdroid/) that are joining Hacktoberfest : https://github.com/TheLastProject/Droidtoberfest)_

**Comment by andrewtavis** [comment]
Hey @shuvashish76 👋 Thanks for reaching out here as well! As said in Matrix, we'll 100% be on F-Droid, and we'll also be in Havktoberfest too! Scribe has been a Hacktoberfest participant for years now 🎃 This will be Scribe-Android's first though. @angrezichatterbox: let's be sure to have as many good first issues up for the start of Hacktoberfest as we can. Maybe we can do a call and map out issues to be made :)

**Comment by angrezichatterbox** [comment]
Yes, we can map out issues. There are some style issues in the app now which I think would be a good addition to good first issues.

**Comment by andrewtavis** [comment]
I think that we're ready to add Scribe-Android to [firstcontributions.github.io](https://github.com/firstcontributions/firstcontributions.github.io), @angrezichatterbox :) Do you want me to send along the PR for it? It would be like [these that I opened for Scribe-iOS](https://github.com/firstcontributions/firstcontributions.github.io/pulls?q=is%3Apr+scribe+is%3Aclosed).

**Comment by angrezichatterbox** [comment]
Yes. Makes sense first contributions is perfect for finding repositories to contribute. We could also add it to this right [Good First Issues](goodfirstissue.dev) .

**Comment by andrewtavis** [comment]
Yes, I can do all of these soon :)

**Comment by andrewtavis** [comment]
Note that I've sent along PRs to the following:
- [firstcontributions.github.io/pull/431](https://github.com/firstcontributions/firstcontributions.github.io/pull/431)
- [up-for-grabs.net/pull/4816](https://github.com/up-for-grabs/up-for-grabs.net/pull/4816)

**Comment by angrezichatterbox** [comment]
Have we ever thought of having a LinkedIn Profile for Scribe @andrewtavis. Some open source orgs have it and it could open us to a lot of new developers.

**Comment by andrewtavis** [comment]
CC @mhmohona who has brought up the same. As you all might have noticed, I'm not a LinkedIn user, but I'm happy to set something up if need be. It might make sense that @mhmohona makes it and makes @angrezichatterbox an owner as well (not sure how it works over there)? I'm fine with all org members having whatever level of admin rights and we can make sure there are enough owners that we have solid coverage?

**Comment by angrezichatterbox** [comment]
> CC @mhmohona who has brought up the same. As you all might have noticed, I'm not a LinkedIn user, but I'm happy to set something up if need be. It might make sense that @mhmohona makes it and makes @angrezichatterbox an owner as well (not sure how it works over there)? I'm fine with all org members having whatever level of admin rights and we can make sure there are enough owners that we have solid coverage?

I recently began using LinkedIn and noticed that several organizations have LinkedIn pages where they showcase their projects and accomplishments.

**Comment by mhmohona** [comment]
Hello @andrewtavis @angrezichatterbox, so how about having a LinkedIn page? 
I havent done it before, so not sure about the requirement, shall I open the page in a call with both of you? Or it can be done in next dev sync? 
Will it be a waste of time doing it in dev sync?

**Comment by angrezichatterbox** [comment]
> Hello @andrewtavis @angrezichatterbox, so how about having a LinkedIn page?
I agree with the idea about having a LinkedIn page. 

> havent done it before, so not sure about the requirement, shall I open the page in a call with both of you? Or it can be done in next dev sync?

I am totally fine with having a call for it to be done or discussing it in the next sync

**Comment by andrewtavis** [comment]
Let's take some time in the sync for it 😊

**Comment by Roshanjossey** [comment]
Added to [awesome-for-beginners](https://github.com/MunGell/awesome-for-beginners/pull/1537)

**Comment by meerhaasham1** [comment]
hi @andrewtavis
I would like to take this on
just let me know how you'd like to proceed!

**Comment by andrewtavis** [comment]
Hey @meerhaasham1 👋 This isn't really an issue that we'd assign to someone as it's more just logging the efforts that we've done for outreach as a community :) Do you have some suggestions on what we can do? I'd wait on really pushing the app until it's released, but that doesn't mean that we can't gather ideas 😊

**Comment by meerhaasham1** [comment]
Got it! @andrewtavis   👋 I'll keep that in mind. I'll let you know my thoughts and suggestions on how we can promote Scribe-Android once it’s closer to release, and we can brainstorm together! 😊

**Comment by meerhaasham1** [comment]
> Hey [@meerhaasham1](https://github.com/meerhaasham1) 👋 This isn't really an issue that we'd assign to someone as it's more just logging the efforts that we've done for outreach as a community :) Do you have some suggestions on what we can do? I'd wait on really pushing the app until it's released, but that doesn't mean that we can't gather ideas 😊

For promotion, I think we could target communities like F-Droid, relevant subreddits (r/Android, r/Kotlin), and language learning forums. We could also leverage platforms like Product Hunt and podcasts focused on open-source and tech.

**Comment by meerhaasham1** [comment]
hi @andrewtavis 
I was thinking that we should do this to help promote Scribe-Android too:
Run a beta testing or early access campaign – We could share the app on platforms like Beta Family or Reddit communities like r/Android or r/languagelearning and ask people to try it out and give feedback. That would also build some buzz before the full release.
what do you think?

**Comment by shuvashish76** [comment]
BTW it is possible to publish both stable/beta version at F-Droid ;)

**Comment by shuvashish76** [comment]
Alternatively if you don't want betas at official F-Droid repo for now, it is also possible to publish at @IzzySoft 's repo : https://apt.izzysoft.de/fdroid/ (the largest third-party F-Droid repo), to get feedback from foss enthusiasts :) Need a separate issue for such inclusion discussions. Just let him know when ready to attach apk files at Releases.

**Comment by andrewtavis** [comment]
Thanks for the research @meerhaasham1! Totally agree on the suggestions :) And thanks for the info on F-Droid, @shuvashish76! We'll 100% get Scribe-Android onto F-Droid, and that repo would be a good resource for early feedback 😊

**Comment by meerhaasham1** [comment]
> Thanks for the research [@meerhaasham1](https://github.com/meerhaasham1)! Totally agree on the suggestions :) And thanks for the info on F-Droid, [@shuvashish76](https://github.com/shuvashish76)! We'll 100% get Scribe-Android onto F-Droid, and that repo would be a good resource for early feedback 😊

You're welcome! 😊 Glad the suggestions were helpful. Excited to see Scribe-Android heading to F-Droid
Looking forward to contributing more 🚀

**Comment by Sharvari2309-tech** [comment]
Hi everyone 👋 Just went through the discussion — the idea of using @IzzySoft’s repo for early feedback sounds great!

If there’s any help needed in setting up the F-Droid metadata or preparing the APK files for release, I’d love to contribute. I’m trying to get involved in open-source and would be happy to support with documentation or testing too. 😊

Let me know how I can assist!

**Comment by IzzySoft** [comment]
As for metadata in terms of description and graphics, please have a look at the [IzzyOnDroid Fastlane Documentation](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Fastlane). For picking up the proper APKs, we'd need them attached to their corresponding [Releases](https://github.com/scribe-org/Scribe-Android/releases). If you want to split between "some releases go to F-Droid, and **others** to IzzyOnDroid", we'd need proper tag naming patterns to set that up separately. Otherwise, if IoD should just pick up "early birds" in addition to the proper releases, matching can be done on file name patterns (you'd need to check with F-Droid how they'd want to handle that, though).

Seeing you have no single version tagged yet, I guess you're not at F-Droid yet anyway. You might also wish to check with the [IzzyOnDroid App Inclusion Policy](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Inclusion-Policy) to see if they are matched; having no APK there's nothing we can check to tell ourselves, not even on the size limit 😉 

PS: some more hints you can find in my [F-Droid snippet at GitLab](https://gitlab.com/-/snippets/1896503).

**Comment by Sharvari2309-tech** [comment]
@andrewtavis  Thanks! That sounds great. Yes, I’ll go ahead and create an issue describing the required F-Droid metadata — including things like the `metadata.yml`, build parameters, and any dependencies.
Once that’s up, I’ll start working on the PR to implement it.
Let me know if there’s any existing reference (e.g., a similar app in F-Droid) you’d recommend I look at for structure. Will post the issue soon!

**Comment by andrewtavis** [comment]
Thanks so much for all the information, @IzzySoft! And thanks as well for taking the lead on this, @Sharvari2309-tech :) As far as repos that are similar, you can check out the following two, the first of which is the project that was forked to start Scribe-Android, and the second being a fork of the first:

- https://github.com/SimpleMobileTools/Simple-Keyboard
- https://github.com/FossifyOrg/Keyboard

Looks like we should be able to reference the second project for this, but please do make an issue. Feel free to start working on it as soon as you make it. We'll assign as soon as we can 😊

**Comment by IzzySoft** [comment]
Gladly, @andrewtavis – please let me know once there's an APK to check, then I fill those gaps as well. With some luck, we have Scribe listed soon!

**Comment by Sharvari2309-tech** [comment]
Hey  @andrewtavis 👋 sorry for the late reply! I’d love to get started, but I’m a bit unsure about some parts of the metadata process. Would it be okay if I opened the issue with what I understand so far, and then refined it with your feedback?"

**Comment by andrewtavis** [comment]
100%, @Sharvari2309-tech! Would be a perfect next step :)
